contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/wifi_test.py

```diff
@@ -0,0 +1,307 @@
+    finally:
+        del_host_conn = "{} c delete {}".format(manager._command, manager.ssid)
+        try:
+            run_command(sshpass_cmd_gen(ip, user, pwd, del_host_conn))
+            logging.info("Deleted host connection successfully.")
+        except Exception as e:
+            logging.error("Failed to delete host connection: %s", str(e))
+        sys.exit(exit_code)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="WiFi test")
+
+    parser.add_argument(
+        "--type", required=True, default="wifi", help="Connection type."
+    )
+    parser.add_argument(
+        "--mode",
+        choices=["ap", "adhoc"],
+        required=False,
+        help="WiFi mode: ap or adhoc",
+    )
+    parser.add_argument(
+        "--interface", required=True, help="WiFi interface to use"
+    )
+    parser.add_argument("--band", required=True, help="WiFi band to use")
+    parser.add_argument("--channel", required=True, help="WiFi channel to use")
+    parser.add_argument(
+        "--keymgmt", required=False, help="Key management method"
+    )
+    parser.add_argument(
+        "--group", required=False, help="Group key management method"
+    )
+    parser.add_argument(
+        "--ssid",
+        default="".join(
+            [random.choice(string.ascii_letters) for i in range(10)]
+        ),
+        required=False,
+        help="SSID for AP mode",
+    )
+    parser.add_argument("--ssid-pwd", required=False, help="Password for SSID")
+    parser.add_argument(
+        "--peer", required=False, help="MAC address for p2p peer"
+    )
+    parser.add_argument(
+        "--host-ip",
+        required=False,
+        help="IP address of the Host device to connect to"
+        "The HOST is a device to access the DUT' AP.",
+    )
+    parser.add_argument(
+        "--host-user",
+        required=False,
+        help="Username of the Host device for SSH connection",
+    )
+    parser.add_argument(
+        "--host-pwd",
+        required=False,
+        help="Password of the Host device for SSH connection",
+    )
+    parser.add_argument(
+        "--set-ap-only",
+        action="store_true",
+        help="Setup DUT's AP mode only. Mostly use for debugging.",
+    )
+
+    args = parser.parse_args()
+
+    if not args.set_ap_only:
+        if not args.host_ip or not args.host_user or not args.host_pwd:
+            parser.error(
+                "--host-ip, --host-user, and --host-pwd are required, "
+                "if you are intend to execute auto test."
+            )
+
+    manager = WiFiManager(
+        args.interface,
+        args.type,
+        args.mode,
+        args.band,
+        args.channel,
+        args.keymgmt,
+        args.group,
+        args.peer,
+        args.ssid,
+        args.ssid_pwd,
+    )
+    with manager:
+        if args.keymgmt is not None:
+            manager.set_secured()
+        if not manager.up_conn():
+            raise RuntimeError("Connection initialization failed!")
```

Comment:
I thought thses actions should be placed into the `__enter__` function

Reply:
that make sense

Reply:
I've pushed the fix